src/app/components/PlayerPanel.ts

```diff
@@ -0,0 +1,132 @@
+
+export class PlayerPanel {
+  /**
+   * Creates PlayerPanel based on data given
+   * @param player accepts instance of current player
+   * @param view accepts instance of View componnet
+   */
+  constructor(private player: Player, private view: View) {}
+
+  createPlayerPanel(): HTMLElement {
+    return Render.elementFactory(
+      'div',
+      {
+        className: 'player-panel',
+        style: `background-color: ${this.player.theColor};`,
+      },
+      this.createPlayerBoard(),
+      this.createResultWindow(),
+      this.createButtonPanel(),
+    );
+  }
+
+  private createPlayerBoard(): HTMLElement {
+    return Render.elementFactory(
+      'div',
+      {
+        className: 'player-panel__board',
+      },
+      Render.elementFactory(
+        'div',
+        { className: 'player-panel__info' },
+        Render.elementFactory('img', {
+          src: this.player.theAvatar,
+          alt: `${this.player.theName}-avatar`,
+          className: 'avatar-icon',
+        }),
+        this.createPlayerDetails(),
+      ),
+      Render.elementFactory(
+        'div',
+        { id: 'time-left', className: 'player-panel__time' },
+        `Time left: ${this.player.theAvatar}`,
+      ),
+      this.createPlayerHerd(),
+    );
+  }
+
+  private createPlayerDetails(): HTMLElement {
+    return Render.elementFactory(
+      'div',
+      { className: 'player-panel__details' },
+      Render.elementFactory('p', {}, 'Current player:'),
+      Render.elementFactory(
+        'p',
+        { className: 'player-panel__name' },
+        `${this.player.theName}`,
+      ),
+    );
+  }
+
+  private createPlayerHerd(): HTMLElement {
+    return Render.elementFactory(
+      'div',
+      { className: 'player-panel__herd' },
+      ...flatten(this.convertHerd()),
+    );
+  }
+
+  private convertHerd(): HTMLElement[] {
+    return this.player.theHerd.theAnimals.map(([animal, count]) => {
+      return Render.elementFactory(
+        'div',
+        { className: 'resources' },
+        Render.elementFactory('img', {
+          className: 'player-panel__image',
+          alt: `${animal.theName}`,
+          src: `./static/images/avatars/${
+            animal.theName === AnimalNames.BIG_DOG ||
+            animal.theName === AnimalNames.SMALL_DOG
+              ? 'dog'
+              : animal.theName
+          }.png`,
+        }),
+        `x${count}`,
+      );
+    });
+  }
+
+  private createResultWindow(): HTMLElement {
+    return Render.elementFactory(
+      'div',
+      {
+        className: 'player-panel__result',
+      },
+      'Results:',
+    );
+  }
+
+  private createButtonPanel(): HTMLElement {
+    return Render.elementFactory(
+      'div',
+      { className: 'player-panel__buttons' },
+      this.createDiceButton(),
+      this.createExchangeButton(),
+    );
+  }
+
+  private createDiceButton(): HTMLElement {
+    const rollBtn = Render.elementFactory(
+      'button',
+      { className: 'btn' },
+      'Roll a dice',
```

Comment:
We will have at least 2 dice, so 'Roll the dice'

Reply:
good point